### PR TITLE
feat: create APIs for get tasks by date, user, state

### PR DESCRIPTION
**Pull request by shelleychen318**
## Notion ticket link

[TaskService BE - Get Tasks by Date, User, State](https://www.notion.so/uwblueprintexecs/F23-Task-Board-181d7a3269374b4c95cca79e331b87e1?p=a3aa363e3b204c6891499e9302ac2a38&pm=s)

## Implementation description
* Created implementations for GetTasksByStartDate, GetTasksByEndDate, and GetTasksByState

## Steps to test
1. Build, test APIs on Postman.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by selena1108** [comment]
Get Tasks by User is already done in previous ticket (GetTasksByAssignerId, GetTasksByAssigneeId)

File: backend/typescript/services/interfaces/taskService.ts [hunk 1]
```diff
@@ -68,6 +68,30 @@ export interface ITaskService {
...
+   * @param endDate end date
+   * @returns a TaskDTO with task's information
+   * @throws Error if task retrieval fails
+   */
+  getTasksByEndDate(endDate: Date): Promise<TaskDTO[]>;
+
+  /**
+   * Get all tasks by a status
+   * @param status status
+   * @returns a TaskDTO with task's information
```
Comment (KathleenX7) [inline backend/typescript/services/interfaces/taskService.ts]: Shouldn't this be a list of TaskDTOs and maybe include details about what type of TaskDTOs are in the list? Same with line 74 and 82.

**Review by KathleenX7 — APPROVED** [review]

File: backend/typescript/services/interfaces/taskService.ts (hunk 1 above)
Comment (selena1108) [reply backend/typescript/services/interfaces/taskService.ts]: Thanks! Added those changes

**Review by williamtran10 — APPROVED** [review]
LGTM, rename the PR using conventional commits naming scheme then squash and merge when ready

**Pull request merged**